### DISCUSSION TITLE
[Turborepo] When running `eas build`, `projectRootDirectory` is determined based on the lock file rather than the .git folder.

### DISCUSSION THREAD

**Discussion by shkdev94** [expo/expo#33339] [Proposals and Ideas]
I'm using Turborepo in a monorepo setup. For optimization, during CI builds, we use `turbo prune` to generate an out directory containing only the required target packages. 

However, when running `eas build` inside the out directory, I encounter a `package.json not found` error. This issue arises because `eas build` determines `projectRootDirectory` based on the .git folder, causing it to reference the wrong path.

I  think it would be better to use a lock file like `package-lock.json` as the reference point for `projectRootDirectory` instead of the .git folder.

Thank you.

**Comment by doug-shontz** [comment]
@boh001 we are seeing the same thing.  What did you do to get around this error?

**Comment by a29paul** [comment]
So this issues happened for me because I had the `out/` in my `.gitignore`. EAS by default also will ignore everything in your `.gitignore`. 

Once I removed that, the build worked. I'll be switching to use a `.easignore` which takes precedence over the `.gitignore` to work around this. See [here](https://docs.expo.dev/build-reference/easignore/) for more details.
